services/github/github.go

```diff
@@ -735,24 +742,37 @@ func (s *Service) Register(oldService types.Service, client types.MatrixClient)
 	return nil
 }
 
-// defaultRepo returns the default repo for the given room, or an empty string.
-func (s *Service) defaultRepo(roomID id.RoomID) string {
-	logger := log.WithFields(log.Fields{
-		"room_id":     roomID,
-		"bot_user_id": s.ServiceUserID(),
-	})
+func (s *Service) loadBotOptions(roomID id.RoomID, logger *log.Entry) (result map[string]interface{}, err error) {
 	opts, err := database.GetServiceDB().LoadBotOptions(s.ServiceUserID(), roomID)
 	if err != nil {
-		if err != sql.ErrNoRows {
-			logger.WithError(err).Error("Failed to load bot options")
+		if err == sql.ErrNoRows {
+			logger.Info("no bot options")
+			return make(map[string]interface{}), nil
+		} else {
+			err := errors.New("Failed to load bot options")
+			logger.WithError(err).Error(err)
+			return nil, err
 		}
-		return ""
 	}
 	// Expect opts to look like:
 	// { github: { default_repo: $OWNER_REPO } }
```

Comment:
Not anymore. Bit of a shame there isn't a struct for this as well.

Reply:
open to suggestions about the struct .. the unmarshalling happens at a higher level of abstraction at https://github.com/matrix-org/go-neb/blob/f71f1a078d469333a4e10a934947eaf0e22e0d2e/database/schema.go#L458 .. so I think it's too late to provide a specifically typed struct by the time we get here..? without a decent refactor